### PR TITLE
List targets quietly

### PR DESCRIPTION
**Pull request by baronfel**
This PR fixes a build break I got while building the repo, and silences the output of the warnings if the `RunContext` verbosity is set to Silent.

### CODE REVIEW

**Comment by baronfel** [comment]
Result after this change:
```
bash-3.2$ dotnet ~/oss/FAKE/src/app/Fake.netcore/bin/Debug/netcoreapp2.0/fake.dll -s build --list
The following targets are available:
   AssemblyInfo
   KillStarted
   Clean
   YarnInstall
   DotnetRestore
   DotnetBuild
   DotnetTest
   WatchUnitTests
   BrunchProduction
   DotnetPublish
   BuildWebsiteDebug
   BuildCollectorDebug
   BuildBrunch
   WatchWebsite
   RunWebsite
   CheckGitlab
   ReleaseGitlab
   DebianPackage
   CopyDeployScripts
   CopyDeployScriptsDebug
   CreateUpdatePackage
   CreateUpdatePackageDebug
   UploadToUpdateServer
   Debug
   Release
   JavascriptBuild
   Build
   RunTests
   Version
   Xml
   BuildProduction
```
which any completion script could easily skip the first line of.
I'd looked at disabling that first line of output as well, but the RunContext isn't something that Targets have access to from what I can tell?

**Comment by baronfel** [comment]
Added sorting of the target names, but it's harder to test that because my script is using proper nugets.

File: src/app/Fake.Runtime/FakeRuntime.fs [hunk 1]
```diff
@@ -611,7 +611,8 @@ let prepareAndRunScript (config:FakeConfig) : RunResult =
 let retrieveHints (config:FakeConfig) (runResult:Runners.RunResult) =
     match runResult with
     | Runners.RunResult.SuccessRun _ -> []
-    | Runners.RunResult.CompilationError err -> [
```
Comment (matthid) [inline src/app/Fake.Runtime/FakeRuntime.fs]: Actually that syntax will no longer yield a warning with next F# version ;)

File: src/app/Fake.netcore/Program.fs [hunk 2]
```diff
@@ -156,7 +156,7 @@ let runOrBuild (args : RunArguments) =
       let result =
         match runResult with
         | Runners.RunResult.SuccessRun warnings ->
-          if warnings <> "" then
+          if warnings <> "" && args.VerboseLevel.PrintNormal then
             traceFAKE "%O" warnings
```
Comment (matthid) [inline src/app/Fake.netcore/Program.fs]: Actually shouldn't this print into standard-error anyway? So your tool can just ignore stderr?

Comment (baronfel) [reply src/app/Fake.netcore/Program.fs]: that's a good point, let me try redirection and see what I get

Comment (baronfel) [reply src/app/Fake.netcore/Program.fs]: Yes you're right of course :D

Comment (baronfel) [reply src/app/Fake.netcore/Program.fs]: we can roll back this change then, I think.

Comment (matthid) [reply src/app/Fake.netcore/Program.fs]: Yes, this basically means it already works are you are not blocked, correct (besides the sort)?

**Comment by baronfel** [comment]
we may not even need the sorting, at least for zsh, because it seems to have sorted it itself:
image

File: src/app/Fake.netcore/Program.fs (hunk 2 above)
Comment (matthid) [reply src/app/Fake.netcore/Program.fs]: Just remove that line change, please

**Comment by matthid** [comment]
Thanks! I think travis is just choking right now as all builds seem to be red right now.

**Pull request merged**